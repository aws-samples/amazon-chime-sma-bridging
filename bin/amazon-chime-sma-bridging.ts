//  “Copyright Amazon.com Inc. or its affiliates.” 
import 'source-map-support/register';
import * as cdk from '@aws-cdk/core';
import { SmaBridgingDemo } from '../lib/amazon-chime-sma-bridging-stack';

const app = new cdk.App();
new SmaBridgingDemo(app, 'SmaBridgingDemo', {});
