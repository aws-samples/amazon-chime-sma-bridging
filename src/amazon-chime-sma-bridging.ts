import { App, Stack, StackProps, CfnOutput } from 'aws-cdk-lib';
import { Construct } from 'constructs';
import { Database, Storage, PSTNAudio } from './index';

export class AmazonChimeSDKPSTNAudioBridging extends Stack {
  constructor(scope: Construct, id: string, props: StackProps = {}) {
    super(scope, id, props);

    const database = new Database(this, 'database');
    const storage = new Storage(this, 'storage');
    const pstnAudio = new PSTNAudio(this, 'pstnAudio', {
      callInfoTable: database.callInfoTable,
      wavFileBucket: storage.wavFileBucket,
    });

    new CfnOutput(this, 'phoneNumber', { value: pstnAudio.inboundPhoneNumber });
  }
}
const devEnv = {
  account: process.env.CDK_DEFAULT_ACCOUNT,
  region: process.env.CDK_DEFAULT_REGION,
};

const app = new App();

new AmazonChimeSDKPSTNAudioBridging(app, 'AmazonChimeSDKPSTNAudioBridging', {
  env: devEnv,
});

app.synth();
