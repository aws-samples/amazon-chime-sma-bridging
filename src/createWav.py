#  “Copyright Amazon.com Inc. or its affiliates.”
import json
import boto3
import os
import wave


wavFileBucket = os.environ["WAVFILE_BUCKET"]

polly = boto3.client("polly")


def createPolly(accountId, id):
    response = polly.synthesize_speech(OutputFormat="pcm", Text=accountId, SampleRate="8000", VoiceId="Joanna")

    if "AudioStream" in response:
        outputWav = id + ".wav"
        with wave.open("/tmp/" + outputWav, "wb") as wav_file:
            wav_file.setparams((1, 2, 8000, 0, "NONE", "NONE"))
            wav_file.writeframes(response["AudioStream"].read())

    return outputWav


def uploadWav(wavFile):
    s3 = boto3.client("s3")
    s3.upload_file("/tmp/" + wavFile, wavFileBucket, wavFile, ExtraArgs={"ContentType": "audio/wav"})

    return


def lambda_handler(event, context):
    print(event)
    if event["Records"][0]["eventName"] != "REMOVE":
        accountId = event["Records"][0]["dynamodb"]["NewImage"]["accountId"]["S"]
        id = event["Records"][0]["dynamodb"]["NewImage"]["id"]["S"]

        accountIdStr = ""
        for x in accountId:
            accountIdStr = accountIdStr + " " + x

        accountIdWav = createPolly(accountIdStr, id)
        uploadWav(accountIdWav)

    return {
        "statusCode": 200,
        "headers": {
            "Content-Type": "application/json",
            "Access-Control-Allow-Origin": "*",
            "Access-Control-Allow-Methods": "OPTIONS,POST",
        },
        "body": json.dumps("Success"),
    }
