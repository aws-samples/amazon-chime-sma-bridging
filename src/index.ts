export * from './database';
export * from './pstnAudio';
export * from './storage';
