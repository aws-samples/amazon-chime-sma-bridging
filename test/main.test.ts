import { App } from 'aws-cdk-lib';
import { Template } from 'aws-cdk-lib/assertions';
import { AmazonChimeSDKPSTNAudioBridging } from '../src/amazon-chime-sma-bridging';

test('Snapshot', () => {
  const app = new App();
  const stack = new AmazonChimeSDKPSTNAudioBridging(app, 'test');

  const template = Template.fromStack(stack);
  expect(template.toJSON()).toMatchSnapshot();
});
