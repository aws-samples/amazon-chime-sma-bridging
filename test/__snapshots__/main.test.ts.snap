// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot 1`] = `
{
  "Outputs": {
    "phoneNumber": {
      "Value": {
        "Fn::GetAtt": [
          "pstnAudioinboundPhoneNumberChimePhoneNumberpstnCustomResourceBC5CD7BC",
          "phoneNumber",
        ],
      },
    },
  },
  "Parameters": {
    "BootstrapVersion": {
      "Default": "/cdk-bootstrap/hnb659fds/version",
      "Description": "Version of the CDK Bootstrap resources in this environment, automatically retrieved from SSM Parameter Store. [cdk:skip]",
      "Type": "AWS::SSM::Parameter::Value<String>",
    },
  },
  "Resources": {
    "CustomCDKBucketDeployment8693BB64968944B69AAFB0CC9EB8756C81C01536": {
      "DependsOn": [
        "CustomCDKBucketDeployment8693BB64968944B69AAFB0CC9EB8756CServiceRoleDefaultPolicy88902FDF",
        "CustomCDKBucketDeployment8693BB64968944B69AAFB0CC9EB8756CServiceRole89A01265",
      ],
      "Properties": {
        "Code": {
          "S3Bucket": {
            "Fn::Sub": "cdk-hnb659fds-assets-\${AWS::AccountId}-\${AWS::Region}",
          },
          "S3Key": "9eb41a5505d37607ac419321497a4f8c21cf0ee1f9b4a6b29aa04301aea5c7fd.zip",
        },
        "Environment": {
          "Variables": {
            "AWS_CA_BUNDLE": "/etc/pki/ca-trust/extracted/pem/tls-ca-bundle.pem",
          },
        },
        "Handler": "index.handler",
        "Layers": [
          {
            "Ref": "storagewavFileBucketDeploymentAwsCliLayer4DCDBF29",
          },
        ],
        "Role": {
          "Fn::GetAtt": [
            "CustomCDKBucketDeployment8693BB64968944B69AAFB0CC9EB8756CServiceRole89A01265",
            "Arn",
          ],
        },
        "Runtime": "python3.9",
        "Timeout": 900,
      },
      "Type": "AWS::Lambda::Function",
    },
    "CustomCDKBucketDeployment8693BB64968944B69AAFB0CC9EB8756CServiceRole89A01265": {
      "Properties": {
        "AssumeRolePolicyDocument": {
          "Statement": [
            {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": {
                "Service": "lambda.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": [
          {
            "Fn::Join": [
              "",
              [
                "arn:",
                {
                  "Ref": "AWS::Partition",
                },
                ":iam::aws:policy/service-role/AWSLambdaBasicExecutionRole",
              ],
            ],
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
    "CustomCDKBucketDeployment8693BB64968944B69AAFB0CC9EB8756CServiceRoleDefaultPolicy88902FDF": {
      "Properties": {
        "PolicyDocument": {
          "Statement": [
            {
              "Action": [
                "s3:GetObject*",
                "s3:GetBucket*",
                "s3:List*",
              ],
              "Effect": "Allow",
              "Resource": [
                {
                  "Fn::Join": [
                    "",
                    [
                      "arn:",
                      {
                        "Ref": "AWS::Partition",
                      },
                      ":s3:::",
                      {
                        "Fn::Sub": "cdk-hnb659fds-assets-\${AWS::AccountId}-\${AWS::Region}",
                      },
                    ],
                  ],
                },
                {
                  "Fn::Join": [
                    "",
                    [
                      "arn:",
                      {
                        "Ref": "AWS::Partition",
                      },
                      ":s3:::",
                      {
                        "Fn::Sub": "cdk-hnb659fds-assets-\${AWS::AccountId}-\${AWS::Region}",
                      },
                      "/*",
                    ],
                  ],
                },
              ],
            },
            {
              "Action": [
                "s3:GetObject*",
                "s3:GetBucket*",
                "s3:List*",
                "s3:DeleteObject*",
                "s3:PutObject",
                "s3:PutObjectLegalHold",
                "s3:PutObjectRetention",
                "s3:PutObjectTagging",
                "s3:PutObjectVersionTagging",
                "s3:Abort*",
              ],
              "Effect": "Allow",
              "Resource": [
                {
                  "Fn::GetAtt": [
                    "storagewavFileBucket13861E9F",
                    "Arn",
                  ],
                },
                {
                  "Fn::Join": [
                    "",
                    [
                      {
                        "Fn::GetAtt": [
                          "storagewavFileBucket13861E9F",
                          "Arn",
                        ],
                      },
                      "/*",
                    ],
                  ],
                },
              ],
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "CustomCDKBucketDeployment8693BB64968944B69AAFB0CC9EB8756CServiceRoleDefaultPolicy88902FDF",
        "Roles": [
          {
            "Ref": "CustomCDKBucketDeployment8693BB64968944B69AAFB0CC9EB8756CServiceRole89A01265",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "CustomPSTNResourcesCustomResourceProviderHandlerD046A72D": {
      "DependsOn": [
        "CustomPSTNResourcesCustomResourceProviderRole536C0E76",
      ],
      "Properties": {
        "Code": {
          "S3Bucket": {
            "Fn::Sub": "cdk-hnb659fds-assets-\${AWS::AccountId}-\${AWS::Region}",
          },
          "S3Key": "dbe1be550cad3176bc39a2ad51044ad674d6b4f0d7dabce112bb5f229bfe455a.zip",
        },
        "Handler": "__entrypoint__.handler",
        "MemorySize": 128,
        "Role": {
          "Fn::GetAtt": [
            "CustomPSTNResourcesCustomResourceProviderRole536C0E76",
            "Arn",
          ],
        },
        "Runtime": "nodejs18.x",
        "Timeout": 300,
      },
      "Type": "AWS::Lambda::Function",
    },
    "CustomPSTNResourcesCustomResourceProviderRole536C0E76": {
      "Properties": {
        "AssumeRolePolicyDocument": {
          "Statement": [
            {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": {
                "Service": "lambda.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": [
          {
            "Fn::Sub": "arn:\${AWS::Partition}:iam::aws:policy/service-role/AWSLambdaBasicExecutionRole",
          },
        ],
        "Policies": [
          {
            "PolicyDocument": {
              "Statement": [
                {
                  "Action": [
                    "chime:CreatePhoneNumberOrder",
                    "chime:CreateSipMediaApplication",
                    "chime:CreateSipRule",
                    "chime:CreateVoiceConnector",
                    "chime:CreateVoiceProfileDomain",
                    "chime:DeletePhoneNumber",
                    "chime:DeleteSipMediaApplication",
                    "chime:DeleteSipRule",
                    "chime:DeleteVoiceConnector",
                    "chime:DeleteVoiceProfileDomain",
                    "chime:GetPhoneNumber",
                    "chime:GetPhoneNumberOrder",
                    "chime:GetSipRule",
                    "chime:ListPhoneNumbers",
                    "chime:PutSipMediaApplicationLoggingConfiguration",
                    "chime:PutVoiceConnectorLoggingConfiguration",
                    "chime:PutVoiceConnectorOrigination",
                    "chime:PutVoiceConnectorStreamingConfiguration",
                    "chime:PutVoiceConnectorTermination",
                    "chime:*MediaInsightsPipelineConfiguration",
                    "chime:SearchAvailablePhoneNumbers",
                    "chime:UpdateSipRule",
                    "chime:AssociatePhoneNumbersWithVoiceConnector",
                    "chime:DisassociatePhoneNumbersFromVoiceConnector",
                  ],
                  "Effect": "Allow",
                  "Resource": "*",
                },
                {
                  "Action": [
                    "logs:DescribeLogGroups",
                    "logs:CreateLogGroup",
                    "logs:CreateLogStream",
                    "logs:ListLogDeliveries",
                    "logs:CreateLogDelivery",
                    "logs:GetLogDelivery",
                    "logs:DeleteLogDelivery",
                  ],
                  "Effect": "Allow",
                  "Resource": "*",
                },
                {
                  "Action": "iam:CreateServiceLinkedRole",
                  "Effect": "Allow",
                  "Resource": "*",
                },
                {
                  "Action": [
                    "lambda:GetPolicy",
                    "lambda:AddPermission",
                  ],
                  "Effect": "Allow",
                  "Resource": "*",
                },
                {
                  "Action": "ssm:*Parameter*",
                  "Effect": "Allow",
                  "Resource": {
                    "Fn::Join": [
                      "",
                      [
                        "arn:aws:ssm:",
                        {
                          "Ref": "AWS::Region",
                        },
                        ":",
                        {
                          "Ref": "AWS::AccountId",
                        },
                        ":parameter/chime/*",
                      ],
                    ],
                  },
                },
              ],
              "Version": "2012-10-17",
            },
            "PolicyName": "Inline",
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
    "CustomS3AutoDeleteObjectsCustomResourceProviderHandler9D90184F": {
      "DependsOn": [
        "CustomS3AutoDeleteObjectsCustomResourceProviderRole3B1BD092",
      ],
      "Properties": {
        "Code": {
          "S3Bucket": {
            "Fn::Sub": "cdk-hnb659fds-assets-\${AWS::AccountId}-\${AWS::Region}",
          },
          "S3Key": "b7f33614a69548d6bafe224d751a7ef238cde19097415e553fe8b63a4c8fd8a6.zip",
        },
        "Description": {
          "Fn::Join": [
            "",
            [
              "Lambda function for auto-deleting objects in ",
              {
                "Ref": "storagewavFileBucket13861E9F",
              },
              " S3 bucket.",
            ],
          ],
        },
        "Handler": "index.handler",
        "MemorySize": 128,
        "Role": {
          "Fn::GetAtt": [
            "CustomS3AutoDeleteObjectsCustomResourceProviderRole3B1BD092",
            "Arn",
          ],
        },
        "Runtime": "nodejs18.x",
        "Timeout": 900,
      },
      "Type": "AWS::Lambda::Function",
    },
    "CustomS3AutoDeleteObjectsCustomResourceProviderRole3B1BD092": {
      "Properties": {
        "AssumeRolePolicyDocument": {
          "Statement": [
            {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": {
                "Service": "lambda.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": [
          {
            "Fn::Sub": "arn:\${AWS::Partition}:iam::aws:policy/service-role/AWSLambdaBasicExecutionRole",
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
    "databasecallInfoTable548E4E70": {
      "DeletionPolicy": "Delete",
      "Properties": {
        "AttributeDefinitions": [
          {
            "AttributeName": "phoneNumber",
            "AttributeType": "S",
          },
        ],
        "BillingMode": "PAY_PER_REQUEST",
        "KeySchema": [
          {
            "AttributeName": "phoneNumber",
            "KeyType": "HASH",
          },
        ],
        "StreamSpecification": {
          "StreamViewType": "NEW_IMAGE",
        },
      },
      "Type": "AWS::DynamoDB::Table",
      "UpdateReplacePolicy": "Delete",
    },
    "pstnAudioemulatorSMALambda96756A02": {
      "DependsOn": [
        "pstnAudiosmaHandlerRoleDefaultPolicy84E51611",
        "pstnAudiosmaHandlerRoleFE1BAD63",
      ],
      "Properties": {
        "Architectures": [
          "arm64",
        ],
        "Code": {
          "S3Bucket": {
            "Fn::Sub": "cdk-hnb659fds-assets-\${AWS::AccountId}-\${AWS::Region}",
          },
          "S3Key": "7b4d8328d248e717610280f94dc24898fdfa7f146af3eaba11ef1bb666b03392.zip",
        },
        "Environment": {
          "Variables": {
            "AWS_NODEJS_CONNECTION_REUSE_ENABLED": "1",
            "CALLINFO_TABLE_NAME": {
              "Ref": "databasecallInfoTable548E4E70",
            },
            "WAVFILE_BUCKET": {
              "Ref": "storagewavFileBucket13861E9F",
            },
          },
        },
        "Handler": "index.handler",
        "Role": {
          "Fn::GetAtt": [
            "pstnAudiosmaHandlerRoleFE1BAD63",
            "Arn",
          ],
        },
        "Runtime": "nodejs16.x",
        "Timeout": 60,
      },
      "Type": "AWS::Lambda::Function",
    },
    "pstnAudioemulatorSipMediaAppsipMediaAppRequestpstnCustomResourceF17FE5F6": {
      "DeletionPolicy": "Delete",
      "Properties": {
        "ServiceToken": {
          "Fn::GetAtt": [
            "CustomPSTNResourcesCustomResourceProviderHandlerD046A72D",
            "Arn",
          ],
        },
        "properties": {
          "endpoint": {
            "Fn::GetAtt": [
              "pstnAudioemulatorSMALambda96756A02",
              "Arn",
            ],
          },
          "name": "testpstnAudioemulatorSipMediaAppE427E61F",
          "region": {
            "Ref": "AWS::Region",
          },
        },
        "resourceType": "SMA",
        "uid": "testpstnAudioemulatorSipMediaAppE427E61F",
      },
      "Type": "Custom::PSTNResources",
      "UpdateReplacePolicy": "Delete",
    },
    "pstnAudioinboundPhoneNumberChimePhoneNumberpstnCustomResourceBC5CD7BC": {
      "DeletionPolicy": "Delete",
      "Properties": {
        "ServiceToken": {
          "Fn::GetAtt": [
            "CustomPSTNResourcesCustomResourceProviderHandlerD046A72D",
            "Arn",
          ],
        },
        "properties": {
          "phoneCountry": "US",
          "phoneNumberType": "Local",
          "phoneProductType": "SipMediaApplicationDialIn",
          "phoneState": "IL",
        },
        "resourceType": "PhoneNumber",
        "uid": "testpstnAudioinboundPhoneNumberBC8BD49F",
      },
      "Type": "Custom::PSTNResources",
      "UpdateReplacePolicy": "Delete",
    },
    "pstnAudioinboundSipMediaAppsipMediaAppRequestpstnCustomResource2E9E3511": {
      "DeletionPolicy": "Delete",
      "Properties": {
        "ServiceToken": {
          "Fn::GetAtt": [
            "CustomPSTNResourcesCustomResourceProviderHandlerD046A72D",
            "Arn",
          ],
        },
        "properties": {
          "endpoint": {
            "Fn::GetAtt": [
              "pstnAudiosmaHandlerLambda397BBDEF",
              "Arn",
            ],
          },
          "name": "testpstnAudioinboundSipMediaApp4F8364C6",
          "region": {
            "Ref": "AWS::Region",
          },
        },
        "resourceType": "SMA",
        "uid": "testpstnAudioinboundSipMediaApp4F8364C6",
      },
      "Type": "Custom::PSTNResources",
      "UpdateReplacePolicy": "Delete",
    },
    "pstnAudioinboundSipRulesipRuleRequestpstnCustomResourceBA975F10": {
      "DeletionPolicy": "Delete",
      "Properties": {
        "ServiceToken": {
          "Fn::GetAtt": [
            "CustomPSTNResourcesCustomResourceProviderHandlerD046A72D",
            "Arn",
          ],
        },
        "properties": {
          "name": "testpstnAudioinboundSipRuleB4CB7DCD",
          "targetApplications": [
            {
              "priority": 1,
              "region": {
                "Ref": "AWS::Region",
              },
              "sipMediaApplicationId": {
                "Fn::GetAtt": [
                  "pstnAudioinboundSipMediaAppsipMediaAppRequestpstnCustomResource2E9E3511",
                  "sipMediaAppId",
                ],
              },
            },
          ],
          "triggerType": "ToPhoneNumber",
          "triggerValue": {
            "Fn::GetAtt": [
              "pstnAudioinboundPhoneNumberChimePhoneNumberpstnCustomResourceBC5CD7BC",
              "phoneNumber",
            ],
          },
        },
        "resourceType": "SMARule",
        "uid": "testpstnAudioinboundSipRuleB4CB7DCD",
      },
      "Type": "Custom::PSTNResources",
      "UpdateReplacePolicy": "Delete",
    },
    "pstnAudiosalesPhoneNumberChimePhoneNumberpstnCustomResourceC8805F3A": {
      "DeletionPolicy": "Delete",
      "DependsOn": [
        "pstnAudioinboundPhoneNumberChimePhoneNumberpstnCustomResourceBC5CD7BC",
      ],
      "Properties": {
        "ServiceToken": {
          "Fn::GetAtt": [
            "CustomPSTNResourcesCustomResourceProviderHandlerD046A72D",
            "Arn",
          ],
        },
        "properties": {
          "phoneCountry": "US",
          "phoneNumberType": "Local",
          "phoneProductType": "SipMediaApplicationDialIn",
          "phoneState": "IL",
        },
        "resourceType": "PhoneNumber",
        "uid": "testpstnAudiosalesPhoneNumberF3B94226",
      },
      "Type": "Custom::PSTNResources",
      "UpdateReplacePolicy": "Delete",
    },
    "pstnAudiosalesSipRulesipRuleRequestpstnCustomResource997959BE": {
      "DeletionPolicy": "Delete",
      "Properties": {
        "ServiceToken": {
          "Fn::GetAtt": [
            "CustomPSTNResourcesCustomResourceProviderHandlerD046A72D",
            "Arn",
          ],
        },
        "properties": {
          "name": "testpstnAudiosalesSipRuleD09EC355",
          "targetApplications": [
            {
              "priority": 1,
              "region": {
                "Ref": "AWS::Region",
              },
              "sipMediaApplicationId": {
                "Fn::GetAtt": [
                  "pstnAudioemulatorSipMediaAppsipMediaAppRequestpstnCustomResourceF17FE5F6",
                  "sipMediaAppId",
                ],
              },
            },
          ],
          "triggerType": "ToPhoneNumber",
          "triggerValue": {
            "Fn::GetAtt": [
              "pstnAudiosalesPhoneNumberChimePhoneNumberpstnCustomResourceC8805F3A",
              "phoneNumber",
            ],
          },
        },
        "resourceType": "SMARule",
        "uid": "testpstnAudiosalesSipRuleD09EC355",
      },
      "Type": "Custom::PSTNResources",
      "UpdateReplacePolicy": "Delete",
    },
    "pstnAudiosmaHandlerLambda397BBDEF": {
      "DependsOn": [
        "pstnAudiosmaHandlerRoleDefaultPolicy84E51611",
        "pstnAudiosmaHandlerRoleFE1BAD63",
      ],
      "Properties": {
        "Architectures": [
          "arm64",
        ],
        "Code": {
          "S3Bucket": {
            "Fn::Sub": "cdk-hnb659fds-assets-\${AWS::AccountId}-\${AWS::Region}",
          },
          "S3Key": "450492823b610a516be3417bc8871e9158b4d395a5ae9e714f086ce8f1e6cfaf.zip",
        },
        "Environment": {
          "Variables": {
            "AWS_NODEJS_CONNECTION_REUSE_ENABLED": "1",
            "CALLINFO_TABLE_NAME": {
              "Ref": "databasecallInfoTable548E4E70",
            },
            "SALES_PHONE_NUMBER": {
              "Fn::GetAtt": [
                "pstnAudiosalesPhoneNumberChimePhoneNumberpstnCustomResourceC8805F3A",
                "phoneNumber",
              ],
            },
            "SUPPORT_PHONE_NUMBER": {
              "Fn::GetAtt": [
                "pstnAudiosupportPhoneNumberChimePhoneNumberpstnCustomResourceC0024E13",
                "phoneNumber",
              ],
            },
            "WAVFILE_BUCKET": {
              "Ref": "storagewavFileBucket13861E9F",
            },
          },
        },
        "Handler": "index.handler",
        "Role": {
          "Fn::GetAtt": [
            "pstnAudiosmaHandlerRoleFE1BAD63",
            "Arn",
          ],
        },
        "Runtime": "nodejs16.x",
        "Timeout": 60,
      },
      "Type": "AWS::Lambda::Function",
    },
    "pstnAudiosmaHandlerRoleDefaultPolicy84E51611": {
      "Properties": {
        "PolicyDocument": {
          "Statement": [
            {
              "Action": [
                "dynamodb:BatchGetItem",
                "dynamodb:GetRecords",
                "dynamodb:GetShardIterator",
                "dynamodb:Query",
                "dynamodb:GetItem",
                "dynamodb:Scan",
                "dynamodb:ConditionCheckItem",
                "dynamodb:BatchWriteItem",
                "dynamodb:PutItem",
                "dynamodb:UpdateItem",
                "dynamodb:DeleteItem",
                "dynamodb:DescribeTable",
              ],
              "Effect": "Allow",
              "Resource": [
                {
                  "Fn::GetAtt": [
                    "databasecallInfoTable548E4E70",
                    "Arn",
                  ],
                },
                {
                  "Ref": "AWS::NoValue",
                },
              ],
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "pstnAudiosmaHandlerRoleDefaultPolicy84E51611",
        "Roles": [
          {
            "Ref": "pstnAudiosmaHandlerRoleFE1BAD63",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "pstnAudiosmaHandlerRoleFE1BAD63": {
      "Properties": {
        "AssumeRolePolicyDocument": {
          "Statement": [
            {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": {
                "Service": "lambda.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": [
          {
            "Fn::Join": [
              "",
              [
                "arn:",
                {
                  "Ref": "AWS::Partition",
                },
                ":iam::aws:policy/service-role/AWSLambdaBasicExecutionRole",
              ],
            ],
          },
        ],
        "Policies": [
          {
            "PolicyDocument": {
              "Statement": [
                {
                  "Action": "chime:*",
                  "Effect": "Allow",
                  "Resource": "*",
                },
              ],
              "Version": "2012-10-17",
            },
            "PolicyName": "chimePolicy",
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
    "pstnAudiosupportPhoneNumberChimePhoneNumberpstnCustomResourceC0024E13": {
      "DeletionPolicy": "Delete",
      "DependsOn": [
        "pstnAudiosalesPhoneNumberChimePhoneNumberpstnCustomResourceC8805F3A",
      ],
      "Properties": {
        "ServiceToken": {
          "Fn::GetAtt": [
            "CustomPSTNResourcesCustomResourceProviderHandlerD046A72D",
            "Arn",
          ],
        },
        "properties": {
          "phoneCountry": "US",
          "phoneNumberType": "Local",
          "phoneProductType": "SipMediaApplicationDialIn",
          "phoneState": "IL",
        },
        "resourceType": "PhoneNumber",
        "uid": "testpstnAudiosupportPhoneNumber2C57B954",
      },
      "Type": "Custom::PSTNResources",
      "UpdateReplacePolicy": "Delete",
    },
    "pstnAudiosupportSipRulesipRuleRequestpstnCustomResourceE4B75E07": {
      "DeletionPolicy": "Delete",
      "Properties": {
        "ServiceToken": {
          "Fn::GetAtt": [
            "CustomPSTNResourcesCustomResourceProviderHandlerD046A72D",
            "Arn",
          ],
        },
        "properties": {
          "name": "testpstnAudiosupportSipRule84B729BA",
          "targetApplications": [
            {
              "priority": 1,
              "region": {
                "Ref": "AWS::Region",
              },
              "sipMediaApplicationId": {
                "Fn::GetAtt": [
                  "pstnAudioemulatorSipMediaAppsipMediaAppRequestpstnCustomResourceF17FE5F6",
                  "sipMediaAppId",
                ],
              },
            },
          ],
          "triggerType": "ToPhoneNumber",
          "triggerValue": {
            "Fn::GetAtt": [
              "pstnAudiosupportPhoneNumberChimePhoneNumberpstnCustomResourceC0024E13",
              "phoneNumber",
            ],
          },
        },
        "resourceType": "SMARule",
        "uid": "testpstnAudiosupportSipRule84B729BA",
      },
      "Type": "Custom::PSTNResources",
      "UpdateReplacePolicy": "Delete",
    },
    "storagewavFileBucket13861E9F": {
      "DeletionPolicy": "Delete",
      "Properties": {
        "PublicAccessBlockConfiguration": {
          "BlockPublicAcls": true,
          "BlockPublicPolicy": true,
          "IgnorePublicAcls": true,
          "RestrictPublicBuckets": true,
        },
        "Tags": [
          {
            "Key": "aws-cdk:auto-delete-objects",
            "Value": "true",
          },
          {
            "Key": "aws-cdk:cr-owned:eadae2c6",
            "Value": "true",
          },
        ],
      },
      "Type": "AWS::S3::Bucket",
      "UpdateReplacePolicy": "Delete",
    },
    "storagewavFileBucketAutoDeleteObjectsCustomResource134585C3": {
      "DeletionPolicy": "Delete",
      "DependsOn": [
        "storagewavFileBucketPolicy0D218DCB",
      ],
      "Properties": {
        "BucketName": {
          "Ref": "storagewavFileBucket13861E9F",
        },
        "ServiceToken": {
          "Fn::GetAtt": [
            "CustomS3AutoDeleteObjectsCustomResourceProviderHandler9D90184F",
            "Arn",
          ],
        },
      },
      "Type": "Custom::S3AutoDeleteObjects",
      "UpdateReplacePolicy": "Delete",
    },
    "storagewavFileBucketDeploymentAwsCliLayer4DCDBF29": {
      "Properties": {
        "Content": {
          "S3Bucket": {
            "Fn::Sub": "cdk-hnb659fds-assets-\${AWS::AccountId}-\${AWS::Region}",
          },
          "S3Key": "3fb6287214999ddeafa7cd0e3e58bc5144c8678bb720f3b5e45e8fd32f333eb3.zip",
        },
        "Description": "/opt/awscli/aws",
      },
      "Type": "AWS::Lambda::LayerVersion",
    },
    "storagewavFileBucketDeploymentCustomResourceAD22553A": {
      "DeletionPolicy": "Delete",
      "Properties": {
        "DestinationBucketName": {
          "Ref": "storagewavFileBucket13861E9F",
        },
        "Prune": true,
        "ServiceToken": {
          "Fn::GetAtt": [
            "CustomCDKBucketDeployment8693BB64968944B69AAFB0CC9EB8756C81C01536",
            "Arn",
          ],
        },
        "SourceBucketNames": [
          {
            "Fn::Sub": "cdk-hnb659fds-assets-\${AWS::AccountId}-\${AWS::Region}",
          },
        ],
        "SourceObjectKeys": [
          "1f06757e90dfa1b60ab13e29010424dc63af89ac132a5f8a08fcb59f14dc3132.zip",
        ],
        "SystemMetadata": {
          "content-type": "audio/wav",
        },
      },
      "Type": "Custom::CDKBucketDeployment",
      "UpdateReplacePolicy": "Delete",
    },
    "storagewavFileBucketPolicy0D218DCB": {
      "Properties": {
        "Bucket": {
          "Ref": "storagewavFileBucket13861E9F",
        },
        "PolicyDocument": {
          "Statement": [
            {
              "Action": [
                "s3:PutBucketPolicy",
                "s3:GetBucket*",
                "s3:List*",
                "s3:DeleteObject*",
              ],
              "Effect": "Allow",
              "Principal": {
                "AWS": {
                  "Fn::GetAtt": [
                    "CustomS3AutoDeleteObjectsCustomResourceProviderRole3B1BD092",
                    "Arn",
                  ],
                },
              },
              "Resource": [
                {
                  "Fn::GetAtt": [
                    "storagewavFileBucket13861E9F",
                    "Arn",
                  ],
                },
                {
                  "Fn::Join": [
                    "",
                    [
                      {
                        "Fn::GetAtt": [
                          "storagewavFileBucket13861E9F",
                          "Arn",
                        ],
                      },
                      "/*",
                    ],
                  ],
                },
              ],
            },
            {
              "Action": [
                "s3:PutObject",
                "s3:PutObjectAcl",
                "s3:GetObject",
                "s3:ListBucket",
              ],
              "Condition": {
                "ArnLike": {
                  "aws:SourceArn": {
                    "Fn::Join": [
                      "",
                      [
                        "arn:aws:chime:*:",
                        {
                          "Ref": "AWS::AccountId",
                        },
                        ":*",
                      ],
                    ],
                  },
                },
                "StringEquals": {
                  "aws:SourceAccount": {
                    "Ref": "AWS::AccountId",
                  },
                },
              },
              "Effect": "Allow",
              "Principal": {
                "Service": "voiceconnector.chime.amazonaws.com",
              },
              "Resource": [
                {
                  "Fn::Join": [
                    "",
                    [
                      {
                        "Fn::GetAtt": [
                          "storagewavFileBucket13861E9F",
                          "Arn",
                        ],
                      },
                      "/*",
                    ],
                  ],
                },
                {
                  "Fn::GetAtt": [
                    "storagewavFileBucket13861E9F",
                    "Arn",
                  ],
                },
              ],
              "Sid": "AWSChimeMediaCaptureBucketPolicy",
            },
          ],
          "Version": "2012-10-17",
        },
      },
      "Type": "AWS::S3::BucketPolicy",
    },
  },
  "Rules": {
    "CheckBootstrapVersion": {
      "Assertions": [
        {
          "Assert": {
            "Fn::Not": [
              {
                "Fn::Contains": [
                  [
                    "1",
                    "2",
                    "3",
                    "4",
                    "5",
                  ],
                  {
                    "Ref": "BootstrapVersion",
                  },
                ],
              },
            ],
          },
          "AssertDescription": "CDK bootstrap stack version 6 required. Please run 'cdk bootstrap' with a recent version of the CDK CLI.",
        },
      ],
    },
  },
}
`;
